### PR TITLE
Library package & examples

### PR DESCRIPTION
**Pull request by dstapl**
Changing package type to a library instead of a binary. Adding documentation for functions along with examples.

### CODE REVIEW

**Comment by Enselic** [comment]
I'm surprised you found my quick and dirty project! May I ask how you found it?

The only use case I personally care about is

```sh
cargo install hex2bin
# For example
readelf -e /usr/bin/ls | hex2dec
```

so splitting code out into src/lib.rs is fine, but we need to keep the binary.

**Comment by dstapl** [comment]
> I'm surprised you found my quick and dirty project! May I ask how you found it?

Hello! I was just searching on crates.io for some recently updated projects, and yours was near the top of the list. 

> so splitting code out into src/lib.rs is fine, but we need to keep the binary.

No worries, I had a slight feeling that might have been the case. I was considering re-adding a `main.rs` file, which would call the `parse_stdin` function as you had it before.

**Comment by dstapl** [comment]
Finished testing on single files using following commands.
```bash
cargo install hex2bin
# On ELF files
readelf -e filename | hex2dec
# or generic text
cat filename | hex2dec
```
Also fixed unwanted spacing between the parsed output lines caused by `println!`.

**Review by dstapl — COMMENTED** [review]
Extracted majority of code to `lib.rs` and retained the binary's CI capabilities in `main.rs` through `parse_ci`.

File: src/main.rs [hunk 1]
```diff
@@ -1,54 +1,10 @@
...
-            ),
-        ];
-        for test in tests {
-            assert_eq!(hex2dec_line(test.0), test.1);
-        }
-    }
+    let callback = |s| print!("{}", s);
+    let on_error = |e| io::Error::new(io::ErrorKind::InvalidInput, e);
+    // (bool, bool, bool) <-> (skip_parse_errors, stop_on_error, break_nl)
+    parse_ci(callback, on_error, false, true, false);
```
Comment (Enselic) [inline src/main.rs:8]: I'm sorry, but this is not an elegant API. I want something simple and elegant and easy to understand.

I have to be honest with you. hex2dec is just a quick and dirty project of mine. I am already involved in other projects that take my time. I'm not really willing to spend a lot of time reviewing code in my toy project hex2dec.

Sorry for the discouraging words, but I wanted to be frank.

**Pull request closed without merging**